wrappers/services_test.go

```diff
@@ -2877,48 +2877,42 @@ func (s *servicesTestSuite) TestQueryDisabledServices(c *C) {
 	// This will mock the following:
 	// svc 1 will be reported as disabled
 	// svc 2 will be reported as enabled
-	r := testutil.MockCommand(c, "systemctl", `#!/bin/sh
-	if [ "$1" = "--root" ]; then
-		# shifting by 2 also drops the temp dir arg to --root
-	    shift 2
-	fi
-
-	case "$1" in
-		show)
-			case "$3" in 
-			"snap.hello-snap.svc1.service"|"snap.hello-snap.svc2.service")
-				for SVC in $3 $4
-				do
-					echo "Type=notify"
-					echo "Id=$SVC"
-					echo "Names=$SVC"
-					echo "NeedDaemonReload=no"
-					if [ "$SVC" = "snap.hello-snap.svc1.service" ]; then
-						echo "ActiveState=inactive"
-						echo "UnitFileState=disabled"
-					elif [ "$SVC" = "snap.hello-snap.svc2.service" ]; then
-						echo "ActiveState=inactive"
-						echo "UnitFileState=enabled"
-					fi
-					if [ "$SVC" != "$4" ]; then 
-						echo ""
-					fi
-				done
-				exit 0
-				;;
-			*)
-				shift 2
-				echo "unexpected show of service $*"
-				exit 2
-				;;
-			esac
-	        ;;
-	    *)
-	        echo "unexpected op $*"
-	        exit 2
-	esac
-	`)
-	defer r.Restore()
+	s.systemctlRestorer = systemd.MockSystemctl(func(cmd ...string) ([]byte, error) {
+		s.sysdLog = append(s.sysdLog, cmd)
+		if len(cmd) >= 3 && cmd[0] == "show" {
+			svcNames := cmd[2:]
```

Comment:
I think we could have reused `systemdtest.HandleMockAllUnitsActiveOutput` here? but I might be wrong

Reply:
nice, didn't know about this one